internal/configs/module.go

```diff
@@ -255,21 +255,7 @@ func NewModule(primaryFiles, overrideFiles []*File, call StaticModuleCall, sourc
 		diags = append(diags, fileDiags...)
 	}
 
-	// Process all module calls now that we have the static context
```

Comment:
If we move this out, we should look at how we handle mod.StaticEvaluator.  I don't think it needs to be a field on mod and we can move a lot more into that decode function

Reply:
@ollevche approach here makes sense to me